### ISSUE TITLE
Feature Request:copy or output to file for mame debug console for cmd ‘history' output

### ISSUE THREAD

**Issue by zengfr** [mamedev/mame#6066]
in mame debug i input cmd "history". 
console print cpu asm code. but can not copy out to.

Feature Request:
1 \ can copy console text output.
2\ console text  can auto   output  to file.

**Comment by angelosa** [comment]
In order to output the CPU disassembler you use `trace` command, please refer to the internal debugger help for usage instructions. 
As for copy the console text output: not sure if we really want to copy to an external stdout buffer or just have to deal with a clipboard, waiting for an opinion before actually closing this.

**Comment by happppp** [comment]
Clipboard is more intuitive (hold left click, select text, copy), this would apply to the debug console window and the errorlog window. I was hoping it's done by simply adding "m_supports_cursor = true;" in emu/debug/dvtext.cpp but nope.

**Comment by cuavas** [comment]
You can copy the visible text from debugger views to the clipboard in the macOS debugger (including colours).  The feature is provided by the debugger module itself though, so it would have to be implemented separately for each implementation.

**Comment by zengfr** [comment]
> In order to output the CPU disassembler you use `trace` command, please refer to the internal debugger `help` for usage instructions.
> As for copy the console text output: not sure if we really want to copy to an external stdout buffer or just have to deal with a clipboard, waiting for an opinion before actually closing this.

Thank you

**Comment by angelosa** [comment]
I'm lazy, I don't want to make a new issue so here we go.

**Comment by ajrhacker** [comment]
For the record, the OS X debugger doesn't have native clipboard support either.

**Comment by cuavas** [comment]
> For the record, the OS X debugger doesn't have native clipboard support either.

You can right-click a debugger view and select "Copy Visible" to copy the visible text in the view to the clipboard, or select "Paste" to simulate typing into the view.  It might not be quite “native” but it's still useful.

**Comment by ajrhacker** [comment]
I didn't realize that was a feature. Of course, with the MacBook I have now, right-clicking doesn't work by right-clicking, and I'm generally not in the habit of it.

**Comment by jotego** [comment]
Text cannot be copied from the debugger window to the clipboard on Windows or Linux as of January 2021. I think that feature would be quite useful, and probably easy to implement.

**Comment by cuavas** [comment]
I’ll get to this after the release.

**Comment by KPreston2900050** [comment]
Here is a thought that escaped all of you.
How about Copy all so I don't have to copy ten lines at a time, paste it then advance ten more lines, rinse and repeat 100 times.

The only thing I have seen with the -debug command is you get to watch your code go round and round.
How about, make a snapshot of the fake memory so you can grab an image of a loaded disk.

I have seen MAME/MESS go downhill for years.  Especially since you merged Mame with Mess.  Thanks Kurt....
The Colecovision and ADAM can no longer use the mouse for the trackball which no longer exists.

How about fix the bugs before adding new systems that no one has ever heard about.
If it were up to Kurt Coder you would be emulating a refrigerator by now.

**Comment by happppp** [comment]
Yes, "Copy Visible" is anti-UI. I'm removing your off-topic rant aimed at Curt, it's unacceptable.

**Comment by jotego** [comment]
I don't get that "copy visible" option neither in linux nor MacOS. For me the debug window is "uncopiable".

**Comment by cuavas** [comment]
>I don't get that "copy visible" option neither in linux nor MacOS. For me the debug window is "uncopiable".

Did you right-click on the actual debug view?
![image](https://user-images.githubusercontent.com/9678431/179236488-afb48f47-c1e3-4afe-b59e-402c7e03d47e.png)

It definitely appears for me with the Qt (default on Linux), Win32 (default on Windows) and Cocoa (default on macOS) debuggers.

**Comment by cuavas** [comment]
Qt debugger showing the item in the context menu:
![image](https://user-images.githubusercontent.com/9678431/179236890-6e6820d0-100d-4469-a78b-51628301311a.png)

**Comment by jotego** [comment]
I have tried again. Somehow MacOS is fine now, still nothing in linux.
